apps/ios/Shared/Views/Chat/Group/GroupMemberInfoView.swift

```diff
@@ -139,17 +171,54 @@ struct GroupMemberInfoView: View {
                     do {
                         let member = try await apiRemoveMember(groupInfo.groupId, member.groupMemberId)
                         await MainActor.run {
-                            _ = ChatModel.shared.upsertGroupMember(groupInfo, member)
+                            _ = chatModel.upsertGroupMember(groupInfo, member)
                             dismiss()
                         }
                     } catch let error {
                         logger.error("apiRemoveMember error: \(responseError(error))")
+                        alert = errorAlert(error, "Error removing member")
                     }
                 }
             },
             secondaryButton: .cancel()
         )
     }
+
+    private func changeMemberRoleAlert() -> Alert {
+        Alert(
+            title: Text("Change member role?"),
+            message: Text("Member role will be changed to ") + Text(newRole.text) + (member.memberCurrent ? Text(". All group members will be notified.") : Text(". The member will receive a new invitation.")),
```

Comment:
Maybe it's better to make full message conditional, and also not split it with member role? - It's easier to translate and not miss dot, for example, also we don't know these will be two separate sentences in different languages, or sentences structure.
```suggestion
            message: member.memberCurrent ? Text("Member role will be changed to \(newRole.text). All group members will be notified.") : Text("Member role will be changed to \(newRole.text). The member will receive a new invitation."),
```